services/provider/server/server.go

```diff
@@ -610,358 +608,172 @@ func decodeAndValidateTicket(ticket string, pubKey *rsa.PublicKey) (*services.On
-							ramenMaintenanceModeLabelKey: "Failover",
-							ramenDRFlattenModeLabelKey:   "force",
-						},
-						Data: mustMarshal(&replicationv1alpha1.VolumeReplicationClassSpec{
-							Parameters: map[string]string{
-								"replication.storage.openshift.io/replication-secret-name": provisionerSecretName,
-								"mirroringMode": "snapshot",
-								"flattenMode":   "force",
-								// This is a temporary fix till we get the replication schedule to ocs-operator
-								"schedulingInterval": volumeReplicationClass5mSchedule,
-								"clusterID":          clientProfile,
-							},
-							Provisioner: util.RbdDriverName,
-						}),
-					},
-				)
+						Name:        vgsc.Name,
+						Kind:        "VolumeGroupSnapshotClass",
+						Labels:      vgsc.Labels,
+						Annotations: vgsc.Annotations,
+						Data:        mustMarshal(vgsc.Parameters),
+					})
 			}
 
-			extR = append(extR,
-				&pb.ExternalResource{
-					Name: "ceph-rbd",
-					Kind: "StorageClass",
-					Labels: map[string]string{
-						ramenDRStorageIDLabelKey: storageID,
-					},
-					Data: mustMarshal(rbdStorageClassData),
-				},
-				&pb.ExternalResource{
-					Name: "ceph-rbd",
-					Kind: "VolumeSnapshotClass",
-					Labels: map[string]string{
-						ramenDRStorageIDLabelKey: storageID,
-					},
-					Data: mustMarshal(map[string]string{
-						"csi.storage.k8s.io/snapshotter-secret-name": provisionerSecretName,
-					}),
-				},
-				&pb.ExternalResource{
-					Name: fmt.Sprintf("%s-groupsnapclass", req.StorageClaimName),
-					Kind: "VolumeGroupSnapshotClass",
-					Labels: map[string]string{
-						ramenDRStorageIDLabelKey: storageID,
-					},
-					Data: mustMarshal(map[string]string{
-						"csi.storage.k8s.io/group-snapshotter-secret-name": provisionerSecretName,
-						"clusterID": clientProfile,
-						"pool":      rns.Spec.BlockPoolName,
-					}),
-				},
-				&pb.ExternalResource{
-					Kind: "ClientProfile",
-					Name: "ceph-rbd",
-					Data: mustMarshal(&csiopv1a1.ClientProfileSpec{
-						Rbd: &csiopv1a1.RbdConfigSpec{
-							RadosNamespace: cephRes.Name,
-						},
-					})},
-			)
-
-		case "CephFilesystemSubVolumeGroup":
-			subVolumeGroup := &rookCephv1.CephFilesystemSubVolumeGroup{}
-			err := s.client.Get(ctx, types.NamespacedName{Name: cephRes.Name, Namespace: s.namespace}, subVolumeGroup)
-			if err != nil {
-				return nil, status.Errorf(codes.Internal, "failed to get %s cephFilesystemSubVolumeGroup. %v", cephRes.Name, err)
+		case storagev1.SchemeGroupVersion.WithKind("StorageClass"):
+			storageClass := resource.(*storagev1.StorageClass)
+			if req.StorageClaimName == util.GenerateNameForCephBlockPoolStorageClass(storageCluster) {
+				storageClass.Parameters["radosnamespace"] = consumerConfig.GetRbdRadosNamespaceName()
+			} else if req.StorageClaimName == util.GenerateNameForCephFilesystemStorageClass(storageCluster) {
+				storageClass.Parameters["subvolumegroup"] = consumerConfig.GetSubVolumeGroupName()
```

Comment:
I don't think we need to add the radosnamespace and subvolumegroup field here, we are deleting them on the client side in 4.18 clusters. https://github.com/red-hat-storage/ocs-client-operator/blob/release-4.18/internal/controller/storageclaim_controller.go#L426-L427

Reply:
Its being deleted after the value is consumed right?

Reply:
I don't see the value consumed